src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/BlobBuilder.cs

```diff
@@ -735,7 +735,7 @@ public void WriteBytes(ImmutableArray<byte> buffer)
         /// <exception cref="InvalidOperationException">Builder is not writable, it has been linked with another one.</exception>
         public void WriteBytes(ImmutableArray<byte> buffer, int start, int byteCount)
         {
-            WriteBytes(ImmutableByteArrayInterop.DangerousGetUnderlyingArray(buffer), start, byteCount);
+            WriteBytes(ImmutableByteArrayInterop.DangerousGetUnderlyingArray(buffer)!, start, byteCount);
```

Comment:
Why can `DangerousGetUnderlyingArray` return null? Will the underlying array ever be null?

Reply:
It can, in cases where `ImmutableArray.IsDefault` is true.